Public/Src/Tools/Xldb.Proto/EnumsAndHelperStructs.proto

```diff
@@ -338,176 +350,58 @@ enum PipType{
-    PipGraphAddPipToPipDependency = 13;
-    PipGraphAddServicePipDependency = 14;
-    PipGraphQueryFileArtifactPipDataWriteFile = 15;
-    PipGraphQueryFileArtifactPipDataCopyFile = 16;
-    PipGraphGetProducingPip = 17;
-    PipGraphListSealedDirectoryContents = 18;
-    PipGraphIsValidInputFileArtifact1 = 19;
-    PipGraphIsValidInputFileArtifact2 = 20;
-    PipGraphIsValidInputFileArtifact3 = 21;
-    PipGraphIsValidOutputFileArtifactSealing1 = 22;
-    PipGraphIsValidOutputFileArtifactSealing2 = 23;
-    PipGraphIsValidOutputFileArtifactRewrite1 = 24;
-    PipGraphIsValidOutputFileArtifactRewrite2 = 25;
-    PipGraphIsValidOutputFileArtifactRewrite3 = 26;
-    PipGraphIsValidOutputDirectory1 = 27;
-    PipGraphFilterNodes = 28;
-    SchedulerOnPipCompleted = 29;
-    SchedulerSchedulePipIfReady = 30;
-    SchedulerReportSkippedPip = 31;
-    SchedulerTryMaterializePipDependenciesAsyncCopyFile = 32;
-    SchedulerTryMaterializePipDependenciesAsyncIpcPip = 33;
-    SchedulerInternalTryMaterializePipOutputsAsync = 34;
-    SchedulerGetAndRecordFileContentHashAsync = 35;
-    SchedulerInternalTryComputePipFingerprintAsync1 = 36;
-    SchedulerInternalTryComputePipFingerprintAsync2 = 37;
-    SchedulerInternalTryComputePipFingerprintAsync3 = 38;
-    SchedulerInternalTryComputePipFingerprintAsync4 = 39;
-    SchedulerOnEstablishingPipFingerprintCompleted = 40;
-    SchedulerInternalTryMaterializePipOutputsTopDownAsync = 41;
-    SchedulerOnMaterializingPipOutputs1 = 42;
-    SchedulerOnMaterializingPipOutputs2 = 43;
-    SchedulerOnMaterializingPipOutputs3 = 44;
-    SchedulerInvalidateMaterializedPipOutputs1 = 45;
-    SchedulerInvalidateMaterializedPipOutputs2 = 46;
-    SchedulerInvalidateMaterializedPipOutputs3 = 47;
-    SchedulerFileContentManagerHostMaterializeFile = 48;
-    SchedulerMaterializePipOutputs2 = 49;
-    SchedulerMaterializePipOutputs3 = 50;
-    SchedulerExecutePips = 51;
-    SchedulerReplayProcessWithWarnings = 52;
-    SchedulerOnPipSemaphoreQueued = 53;
-    SchedulerOnPipSemaphoreDequeued = 54;
-    SchedulerInternalTryComputePipFingerprintAsync = 55;
-    SchedulerAreInputsPresentForSkipDependencyBuild = 56;
-    SchedulerPartialGraphReload = 57;
-    SchedulerExecuteSealDirectoryPip = 58;
-    SealedDirectoryTableTryFindDirectoryArtifactContainingFileArtifact = 59;
-    PipViewerRequest = 60;
-    ViewerAnalyzer = 61;
-    LoggingUncacheableProcessImpact = 62;
-    LoggingPipFailedOnWorker = 63;
-    Test = 64;
-    PipQueueGetCpuQueueProcessResources = 65;
-    FileMonitoringViolationAnalyzerClassifyAndReportAggregateViolations = 66;
-    SchedulerCloudBuildLogging = 67;
-    IdeGenerator = 68;
-    RunnablePip = 69;
-    CollectPipInputsToMaterializeForIPC = 70;
-    PipGraphPostValidation = 71;
-    FingerprintStore = 72;
-    PipGraphGetSealDirectoryByKind = 73;
-    GetSealDirectoryFingerprint = 74;
-    Explorer = 75;
-    PreserveOutput = 76;
-    End = 77;
-}
-
 enum Options{

```

Comment:
>Options [](start = 5, length = 7)

what about `IncrementalTool` value? how do you handle it?

Reply:
Let me go ahead and add that.

---
In reply to: [320543502](https://github.com/microsoft/BuildXL/pull/820#discussion_r320543502) [](ancestors = 320543502)